Model/AdminSdkProxy.php

```diff
@@ -0,0 +1,138 @@
+use Rvvup\Payments\Model\SdkProxy;
+use Rvvup\Sdk\GraphQlSdk;
+use Rvvup\Sdk\GraphQlSdkFactory;
+
+class AdminSdkProxy extends SdkProxy
+{
+    /** @var RequestInterface */
+    private $request;
+
+    /** @var OrderRepositoryInterface */
+    private $orderRepository;
+
+    /** @var InvoiceRepositoryInterface */
+    private $invoiceRepository;
+
+    /** @var Registry */
+    private $registry;
+
+    /** @var GraphQlSdk */
+    private $subject;
+
+    /**
+     * @param ConfigInterface $config
+     * @param UserAgentBuilder $userAgent
+     * @param GraphQlSdkFactory $sdkFactory
+     * @param StoreManagerInterface $storeManager
+     * @param GetEnvironmentVersionsInterface $getEnvironmentVersions
+     * @param RequestInterface $request
+     * @param OrderRepositoryInterface $orderRepository
+     * @param InvoiceRepositoryInterface $invoiceRepository
+     * @param Registry $registry
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        ConfigInterface $config,
+        UserAgentBuilder $userAgent,
+        GraphQlSdkFactory $sdkFactory,
+        StoreManagerInterface $storeManager,
+        GetEnvironmentVersionsInterface $getEnvironmentVersions,
+        RequestInterface $request,
+        OrderRepositoryInterface $orderRepository,
+        InvoiceRepositoryInterface $invoiceRepository,
+        Registry $registry,
+        LoggerInterface $logger
+    ) {
+        $this->request = $request;
+        $this->orderRepository = $orderRepository;
+        $this->invoiceRepository = $invoiceRepository;
+        $this->registry = $registry;
+        parent::__construct(
+            $config,
+            $userAgent,
+            $sdkFactory,
+            $storeManager,
+            $getEnvironmentVersions,
+            $logger
+        );
+    }
+
+    /**
+     * @return int|null
+     */
+    private function getStoreIdUsed(): ?int
+    {
+        $storeId = null;
+        $orderId = $this->request->getParam('order_id');
+        if (!$orderId) {
+            $creditMemo = $this->registry->registry('current_creditmemo');
+            $orderId = $creditMemo ? $creditMemo->getOrderId() : null;
+        }
+
+        if ($orderId) {
+            try {
+                $order = $this->orderRepository->get((int) $orderId);
+                $storeId = (int)$order->getStoreId();
+                $this->storeManager->setCurrentStore($storeId);
+            } catch (\Exception $e) {
+                return $storeId;
+            }
+        } elseif ($this->request->getParam('invoice_id')) {
+            $invoiceId = $this->request->getParam('invoice_id');
+            try {
+                $invoice = $this->invoiceRepository->get((int) $invoiceId);
+                $storeId = (int) $invoice->getStoreId();
+                $this->storeManager->setCurrentStore($storeId);
+            } catch (\Exception $e) {
+                return $storeId;
```

Comment:
Is it worth adding error logs on these cases where we can't find the store? Don't think we'd not expect to hit them if things are working correctly?